Source/Diagnostics/Diagnostics.cpp

```diff
@@ -509,27 +504,26 @@ Diagnostics::InitData ()
 
 
 void
-Diagnostics::InitBaseData ()
+Diagnostics::InitBaseData (const InitDiagnosticsParameters& params, [[maybe_unused]] amrex::AmrMesh* p_warpx_mesh)
 {
-    auto & warpx = WarpX::GetInstance();
     // Number of levels in the simulation at the current timestep
-    nlev = warpx.finestLevel() + 1;
+    nlev = params.finest_level + 1;
     // default number of levels to be output = nlev
     nlev_output = nlev;
     // Maximum number of levels that will be allocated in the simulation
-    nmax_lev = warpx.maxLevel() + 1;
+    nmax_lev = params.max_level + 1;
     m_all_field_functors.resize( nmax_lev );
 
     // For restart, move the m_lo and m_hi of the diag consistent with the
     // current moving_window location
-    if (WarpX::do_moving_window) {
-        const int moving_dir = WarpX::moving_window_dir;
+    if (params.do_moving_window) {
+        const int moving_dir = params.moving_window_dir;
         const amrex::Real displacement =
-            warpx.getmoving_window_x() - warpx.Geom(0).ProbLo(moving_dir);
+            params.moving_window_x - warpx.Geom(0).ProbLo(moving_dir);
```

Comment:
@lucafedeli88 I rebased your PR but in the meantime this extra term came in here.

I think you need to store `warpx.Geom(0).ProbLo(moving_dir)` now as well in `InitDiagnosticsParameters`?